File: 338 Counting Bits/Counting Bits.py
#Python3 versiyonunu seçmeyi unutma =)

class Solution:
    def countBits(self, n: int) -> List[int]:
        dp = [0] * (n+1)
        k =1
        for i in range(1 , n+1):

            if k *2 == i:
                k =i

            dp[i] = 1 + dp[i-k]
        return dp

File: 338 Counting Bits/Counting Bits.cpp
#include <vector>

class Solution {
public:
    std::vector<int> countBits(int n) {
        std::vector<int> dp(n + 1, 0);
        int k = 1;
        for (int i = 1; i <= n; i++) {
            if (k * 2 == i) {
                k = i;
            }
            dp[i] = 1 + dp[i - k];
        }
        return dp;
    }
};
